spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -3179,17 +3318,36 @@ def leaveEvent(self, event):
             QApplication.restoreOverrideCursor()
             self.__cursor_changed = False
             self.clear_extra_selections('ctrl_click')
+            self._last_hover_uri = None
         TextEditBaseWidget.leaveEvent(self, event)
 
     def mousePressEvent(self, event):
-        """Reimplement Qt method"""
+        """Override Qt method."""
         ctrl = event.modifiers() & Qt.ControlModifier
         alt = event.modifiers() & Qt.AltModifier
         pos = event.pos()
         if event.button() == Qt.LeftButton and ctrl:
             TextEditBaseWidget.mousePressEvent(self, event)
             cursor = self.cursorForPosition(pos)
-            self.go_to_definition_from_cursor(cursor)
+
+            if self._last_hover_uri:
+                uri = self._last_hover_uri
+                if uri.startswith('file://'):
+                    fname = self._preprocess_file_uri(uri)
+
+                    if osp.isfile(fname) and encoding.is_text_file(fname):
+                        # Open in editor
+                        self.go_to_definition.emit(fname, 0, 0)
+                    else:
+                        # Use external program
+                        quri = QUrl.fromLocalFile(fname)
+                        QDesktopServices.openUrl(quri)

```

Comment:
For local files I'd prefer to use the following (which is what `open_file` in `MainWindow` does too):

```python
fname = file_uri(fname)
programs.start_file(fname)
```

`file_uri` is present in `qthelpers` and `programs` refers to `utils/programs.py`